### ISSUE TITLE
ImageExtra breaking image replacement when Maximum Files Allowed set to 1

### ISSUE THREAD

**Issue by adrianbj** [justb3a/processwire-imageextra#10]
https://processwire.com/talk/topic/9056-module-image-extra/?p=107446

With ImageExtra installed and Maximum Files Allowed set to "1" when I go to upload a second image, the editor actually shows both images. After save, the first one disappears, but without ImageExtra, the second one instantly overwrites the first one. This has been confusing for our users - they think a second image has been uploaded rather than overwriting the first one. I hope you can reproduce at your end.
 
I wonder if it's a simple matter of incorporating some of the new code in: 
https://github.com/ryancramerdesign/ProcessWire/blob/master/wire/modules/Inputfield/InputfieldFile/InputfieldFile.module#L438
 
into your module at:
https://github.com/justonestep/processwire-imageextra/blob/master/ImageExtra.module#L540

**Comment by justonestep** [comment]
> I wonder if it's a simple matter of incorporating some of the new code in: 
> https://github.com/r...ile.module#L438
> 
> into your module at:
> https://github.com/j...tra.module#L540

These links are broken. The last link should be: https://github.com/justonestep/processwire-imageextra/blob/master/ImageExtra.module#L540. Could you please correct the first link?

**Comment by adrianbj** [comment]
Oops - sorry about that - links have been corrected.

**Comment by justonestep** [comment]
I figured out the problem but I've no idea how to fix this. The problem is just this line ([:532](https://github.com/justonestep/processwire-imageextra/blob/develop/ImageExtra.module#L532)). I cannot access / overwrite `$singleFileReplacement` because it's a protected variable ([:46](https://github.com/ryancramerdesign/ProcessWire/blob/master/wire/modules/Inputfield/InputfieldFile/InputfieldFile.module#L46)). It's used to return the ajax request which is also a protected function ([:654](https://github.com/ryancramerdesign/ProcessWire/blob/master/wire/modules/Inputfield/InputfieldFile/InputfieldFile.module#L654)).

**Comment by mn-martin** [comment]
@justonestep This'll work ;-)

```
// imitate $this->singleFileReplacement = true;
$reflectionClass = new ReflectionClass(get_class($event->object));
$reflectionProperty = $reflectionClass->getProperty('singleFileReplacement');
$reflectionProperty->setAccessible(true);
$reflectionProperty->setValue($event->object, true);
```

**Comment by justb3a** [comment]
fixed in latest release (1.0.0)
